api/views/user_views.py

```diff
@@ -718,6 +713,62 @@ def post(self, request):
             )
 
 
+class JWTGoogleLogin(APIView):
+    permission_classes = [AllowAny]
+
+    @extend_schema(
+        request=GoogleLoginRequestSerializer,
+        responses={
+            200: GoogleLoginResponseSerializer,
+            400: OpenApiTypes.OBJECT,
+        },
+        tags=["Authentication Endpoint"],
+    )
+    def post(self, request):
+        token = request.data.get("token")
```

Comment:
Might be good to give this another name; perhaps `google_token`, since this is not the same access/refresh tokens that we generate; we can do that in a subsequent PR.